### ISSUE TITLE
Python 2.7

### ISSUE THREAD

**Issue by Kadantte** [felixrieseberg/windows-build-tools#206]
Why the tool still Install python 2.7 when it needs to install 3.8?
```shell
PS C:\Windows\system32> npm install --global windows-build-tools

> windows-build-tools@5.2.2 postinstall C:\Users\Candyseul\AppData\Roaming\npm\node_modules\windows-build-tools
> node ./dist/index.js



Downloading python-2.7.15.amd64.msi
[============================================>] 100.0% of 20.25 MB (1.12 MB/s)
Downloaded python-2.7.15.amd64.msi. Saved to C:\Users\Candyseul\.windows-build-tools\python-2.7.15.amd64.msi.
Downloading vs_BuildTools.exe
[============================================>] 100.0% of 1.12 MB (746.57 kB/s)
Downloaded vs_BuildTools.exe. Saved to C:\Users\Candyseul\.windows-build-tools\vs_BuildTools.exe.

Starting installation...
Launched installers, now waiting for them to finish.
This will likely take some time - please be patient!

Status from the installers:
---------- Visual Studio Build Tools ----------
Successfully installed Visual Studio Build Tools.
------------------- Python --------------------

Now configuring the Visual Studio Build Tools and Python...

All done!

+ windows-build-tools@5.2.2
added 145 packages from 99 contributors in 232.448s
```

**Comment by MFunction96** [comment]
I have same question, too. I can not install `windows-build-tools` with `Python 3.8`.
In addition, I tried to install `windows-build-tools` by **offline** installation manually. But there is no `python-3.8.1.amd64.msi` or `python-3.8.1.msi` provided by Python official website.

**Comment by basickarl** [comment]
Our Azure CI is blowing up due to this. Trying to install version 5.2.2 but the following happens:
```
> windows-build-tools@5.2.2 postinstall C:\npm\prefix\node_modules\windows-build-tools
> node ./dist/index.js

Downloading python-2.7.15.amd64.msi
[>                                            ] 0.0% (0 B/s)
...
Successfully installed Python 2.7Terminate batch job (Y/N)? 
##[error]The operation was canceled.
Finishing: Install
```
Eventually the CI timeout hits and kasplat.

**Comment by jacobq** [comment]
> Why the tool still Install python 2.7 when it needs to install 3.8?

All but the latest versions of node lack support for Python 3, if I am not mistaken. Also, there hasn't been a release of this module published since the change to use Python 3 landed in `master`.


Regarding the problem, is it that the download finishes, but the process gets stuck after installation?

**Comment by Kadantte** [comment]
@jacobq For me, it installs everything fine but for now I want it to install python 3 instead but since you make it clear it's okay.

**Comment by joginds4** [comment]
Python 2 has already sunset on Jan 1, 2020. Please refer https://www.python.org/doc/sunset-python-2/ so when can we get windows-build-tools with Python 3.x?

**Comment by GeorgesBat** [comment]
> 
> 
> Python 2 has already sunset on Jan 1, 2020. Please refer https://www.python.org/doc/sunset-python-2/ so when can we get windows-build-tools with Python 3.x?

I second this I still cant get windows build tools because I need Python 3.7

**Comment by cclauss** [comment]
@felixrieseberg Please make a release with the current code.  ___Legacy Python is dead___ so this tool needs to stop installing it.

**Comment by yongruning** [comment]
Excuse me，i want to get python3,8 from this tool,what  should i do?

**Comment by cclauss** [comment]
Follow the [Offline Installation](https://github.com/felixrieseberg/windows-build-tools#offline-installation) instructions but change all instances of `3.8.1` to `3.8.2`.

**Comment by yongruning** [comment]
After using this tool,I removed python2.7 from the Windows path,Then installed python3.8 locally and configured it to Windows path,Successfully used

**Comment by cclauss** [comment]
Sure.  These days you can always install the latest Python from the Microsoft Windows Store.  This advantages of that approach is that you do not need to configure and it will be easy to upgrade as new versions of CPython are released.  `Python 3.8.3` is [scheduled to released next month](https://www.python.org/dev/peps/pep-0569/#schedule).

**Comment by yongruning** [comment]
It would be better if I could choose the python version on windows-build-tools. Like vs, I have been struggling with environment setup for a long time，I am a beginner，Because the company has a technical need

**Comment by Cobertos** [comment]
> Follow the [Offline Installation](https://github.com/felixrieseberg/windows-build-tools#offline-installation) instructions but change all instances of `3.8.1` to `3.8.2`.

This does not seem to work for me:

```
PS C:\WINDOWS\system32> npm install --global windows-build-tools --offline-installers="C:\Users\cobertos\Downloads\python-3.8.2-amd64.exe"
npm WARN deprecated request@2.88.2: request has been deprecated, see https://github.com/request/request/issues/3142

> windows-build-tools@5.2.2 postinstall C:\Program Files\nodejs\node_modules\windows-build-tools
> node ./dist/index.js

Offline installation: Offline path C:\Users\cobertos\Downloads\python-3.8.2-amd64.exe was passed, but we could not find python-2.7.15.amd64.msi in that path. Aborting installation now.
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! windows-build-tools@5.2.2 postinstall: `node ./dist/index.js`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the windows-build-tools@5.2.2 postinstall script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\cobertos\AppData\Roaming\npm-cache\_logs\2020-04-10T12_06_14_031Z-debug.log
```

**Comment by Cobertos** [comment]
Ended up just doing the manual installation as specified in `node-gyp` https://github.com/nodejs/node-gyp#option-2

**Comment by Zireael** [comment]
April 2020
Why this still tries to install Python 2.7?

**Comment by Kadantte** [comment]
Well, It's been a long time...

**Comment by aminya** [comment]
The package needs registration. #223 will probably solve this!

**Comment by abdulraheem9909** [comment]
![image](https://user-images.githubusercontent.com/72109121/103016761-63e61000-4564-11eb-8762-890605645d4b.png)

**Comment by abdulraheem9909** [comment]
I am having issue while installing windows-build-tool

**Comment by cclauss** [comment]
Python 2 died 12 months ago on 1/1/2020.  The best path forward is to download the current Python 3.9.1 from the Microsoft App Store.

**Comment by cizario** [comment]
January 2021, 
i already have `python 3.7.3` installed for django projects and other stuffs, but when i installed `windows-build-tools` i got `python 2.7.15` installed ! that's really disappointed me since 2.x branch was officially out of support. 
it would be really appreciated if the package during installation detect if python is already installed if not it just gives you the ability to provide the path to python installation.

**Comment by jacobq** [comment]
> it would be really appreciated if the package during installation detect if python is already installed if not it just gives you the ability to provide the path to python installation so we avoid any conflict between python installations

Thank you for the suggestion. That sounds like a good improvement. I am sorry for your frustration and admit that this project has been neglected for some time now. Perhaps in the future I will be able to lend a hand again but for now others will need to step up (PRs welcome).

**Comment by cclauss** [comment]
The code on master is already Python 3 so we do not need PR but instead we need a release of the current master.  https://github.com/felixrieseberg/windows-build-tools/releases

**Comment by jacobq** [comment]
I don't think the current code detects what is already installed though.

Unfortunately I do not have the npm credentials to make an official
release. I could, however, make a scoped one for people to use in the mean
time.

-Jacob

**Comment by cclauss** [comment]
Scoped is better than nothing.

**Comment by Kadantte** [comment]
@jacobq any news about your scoped version?

**Comment by jacobq** [comment]
> @jacobq any news about your scoped version?

No, sorry. I haven't forgotten but neither have I carved out time to work on this yet. Thanks for the nudge.
Those willing to help clean-up or review, please react to this comment with 👀 .
Those willing to help test, please react to this comment with 🚀 . Once I get started I think that'd be helpful for getting it over the finish line.

**Comment by Kadantte** [comment]
I get everything working after I did it manually :/ but no harm if everything is automated xD
